src/lib/loaders/loaders_without_authentication/vortex.ts

```diff
@@ -14,17 +19,52 @@ export default (opts) => {
     )
   }
 
+  const vortexGraphqlImpersonationLoader = (opts) => async <T = unknown>({
+    query,
+  }: GraphQLArgs): Promise<Record<string, T>> => {
+    const body = JSON.stringify({
+      query,
+    })
+    const response = await fetch(urljoin(VORTEX_API_BASE, "graphql"), {
+      body,
+      headers: {
+        Authorization: `${opts.appToken}`,
```

Comment:
I'm wondering how this loader gets authorized to hit Vortex. Should this be `Bearer ...` ? And also, what is `opts.appToken` - is that a Vortex token?

Reply:
`opts.appToken` is the `X-Xapp-Token` and in this case it always come as vortex token.

Reply:
Isn't that a Gravity thing though? Only Gravity knows about roles and verifying a Gravity-issued xapp token. Does Vortex know/respect those?

Reply:
Nope [check this out](https://github.com/artsy/vortex/blob/d742bcd5999dcd934d71176b863bf257b925d2b0/app/graphql/resolvers/base_resolver.rb#L22)

Reply:
Gotcha

Reply:
I guess I still don't quite understand how Vortex will verify this token though. The `opts.appToken` is either MP's own xapp-token when the incoming request doesn't specify one, _or_ it's the incoming request's one.

So Braze's xapp token flows through, and is a Gravity xapp token. How does Vortex verify that token? I would think this might still need the `setup(appToken)` part to swap a gravity xapp token to a Vortex token.

(sorry for the many questions!)

Reply:
Got it. Does it make more sense to use `config.VORTEX_TOKEN` instead?

Reply:
That's a token set in Metaphysics ENV and MP isn't the `email_provider`. Most likely this needs a `setup(appToken)` type callback like the other loader to swap an incoming gravity xapp token for a vortex one.

Reply:
It also could be that I'm totally not understanding something about Vortex and how it works...I've never used that app before. I would defer to @jonallured or even @joeyAghion on these maybe? Or if we have a little mob session on this to do some knowledge-sharing.

Reply:
https://github.com/artsy/metaphysics/pull/4030 looks to add some of this vortex functionality (which i honestly don't fully understand), so maybe @jonallured is the person